### DISCUSSION TITLE
errors installing pkg-config

### DISCUSSION THREAD

**Discussion by m-vonahn** [Homebrew/discussions#5194] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.11-105-gaeceee
ORIGIN: https://github.com/Homebrew/brew
HEAD: aeOceee74f1570083a2421be2509443d3a4337f
Last commit: 2 hours ago
Core tap JSON: 11 Mar 01:34 UTC
Core cask tap JSON: 11 Mar 01:34 UTC
HOMEBREW PREFIX: /Users/sonahn/homebrew
CELLAR: /Users/svonahn/homebrew/Cellar
HOMEBREW_CASK_OPTS: [1
HOMEBREW MAKE
_JOBS: 10
Homebrew Ruby: 3.1.4 = /Users/svonahn/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
ags
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 = /Library/Developer /CommandLineTools/usr/bin/g
Curl: 8.4.0 = /usr/bin/curl
macOS: 14.3-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers with debugging if you file an issue. If everything you use Homebrew for 1s working fine: please don't worry or file an issue: just ignore this. Thanks!
Warning: Unbrewed dylibs were found in /us/local/11b. if you didn't put them there on purpose they could cause problems when building Homebrew formulae, and may need to be deleted.
Unexpected dyl10S:
/use/local/11b/11bjson-c.2.dy11b
/us/local/11b/11bykpers-1.1. dyl1b
/usr/local/11b/11byublkey.0.dyl1b
/us/local/11b/11byubtkey.dy11b
Warning: Unbrewed static libraries were found in /us/local/11b.
If you didn't put them there on purpose they could cause problems when bufling Homebrew formulae, and may need to be deleted.
Unexpected static libraries:
/us/local/11b/11byubikey.a
Warning: Your Homebrew's prefix is not /opt/homebrew.
Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or
its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request
```

### Description of issue

when trying to do a 
`brew install wget`
I am getting an error when installing the pkg-config dependency

`gatomic.c:464:10: error: incompatible integer to pointer conversion passing 'gsize' (aka 'unsigned long!)
› l-Wint-conversion]
return g
atomic_pointer_xor ((volatile gpointer *) atomic, val);
• gatomic.h: 191:45: note: expanded from macro 'g_atomic_pointer_xor' (gsize)sync_fetch_and_xor ( (atomic),
(val)):`

This works as expected on another computer, this one it does not, any help is appreciated

**Comment by cho-m** [comment]
Beforehand, are you unable to use the default prefix which would avoid source compiling?
> ```
> HOMEBREW PREFIX: /Users/sonahn/homebrew
> ```

Anyway, looking at your post:
> ```
> CLT: 15.3.0.0.1.1708646388
> ```
> ...
> This works as expected on another computer, this one it does not, any help is appreciated

Most likely due to new CLT which probably pulled in some new defaults.

Taking a look upstream, I see someone has reported it: https://gitlab.freedesktop.org/pkg-config/pkg-config/-/issues/81

Bit worried on 3 year gap in upstream commits but hopefully that is because it has been stable. Worst case, Homebrew would need to follow Arch and switch to `pkgconf` instead - https://lists.archlinux.org/pipermail/arch-dev-public/2018-May/029252.html

**Reply by m-vonahn** [reply]
Unfortunately I cannot use the default prefix, work computer.  Homebrew is allowed but must be in user directory

**Comment by fxcoudert** [comment, marked as answer]
Fix proposed in https://github.com/Homebrew/homebrew-core/pull/165878

**Reply by deanpullenesure** [reply]
Can confirm the fix worked for me having seen this issue too. Thank you.

**Reply by m-vonahn** [reply]
Thank you, this fixed the problem for me as well.
